extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -265,6 +269,28 @@ public String deleteExtensionMetadata(final String extensionName) throws Validat
         }
     }
 
+    public String registerExtensionMetadata(final String extensionName, final String path, final String description)
+        throws URISyntaxException, FalconException, IOException{
+        Configuration conf = new Configuration();
+        URI uri = new URI(path);
+        conf.set("fs.default.name", uri.getScheme() + "://" + uri.getAuthority());
+        FileSystem fileSystem =  HadoopClientFactory.get().createFalconFileSystem(uri);
+        FileStatus[] fileStatusReadMe = fileSystem.listStatus(new Path(uri.getPath() + "/README"));
+        PathFilter filter=new PathFilter(){
+            public boolean accept(Path file){
+                return file.getName().endsWith(".jar");
+            }
+        };
+        FileStatus[] fileStatusJar = fileSystem.listStatus(new Path(uri.getPath() + "/libs"), filter);
```

Comment:
Then we need to correct the doc too because in the doc there is no mention of libs/build or libs/runtime.